pkg/stats/collector.go

```diff
@@ -251,18 +255,21 @@ func (s *BufferedCollector) Run(ctx context.Context) {
 	go func() {
 		for {
 			select {
-			case w := <-s.writes: // collect events
+			case w := <-s.writes:
+				// collect events, and flush if needed by size
 				s.incr(w)
+				if len(s.cache) >= s.flushSize {
+					s.flush()
+				}
```

Comment:
maybe I'm missing that, but where does the cache cleanup? I couldn't find the reference in the `flush` function

Reply:
In `flush()` implementation `s.cache = make(keyIndex)` we reset the cache.

Reply:
🤦 sorry, missed that